### PR TITLE
Awesome Books created with Javascript classes

### PR DESCRIPTION
**Pull request by Rachelwebdev**
### In this milestone, we

- created JavaScript class methods to add new books to the list
-  created JavaScript class methods to remove new books from the list
- Styled the booklist to match the wireframe
- Fixed Linter errors

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,52 +1,68 @@
...
-  li.insertBefore(deleteBtn, li.lastElementChild);
-  booksList.appendChild(li);
-  deleteBtn.addEventListener('click', () => {
-    books = books.filter((books) => {
-      if (books.title !== title && books.author !== author) {
+  addBook() {
+    const { title, author } = this;
+    const bookObj = { title, author };
+    if (title !== '' || author !== '') {
+      books.push(bookObj);
```
Comment (rloterh) [inline script.js:16]: - [ ] OPTIONAL
It's a good idea to use non mutating array methods `concat`, over `push`, so as to keep the original array intact.

Comment (rloterh) [inline script.js]: - [ ] Your app happens to break after refactoring your code. An error is thrown in an attempt to add new book details. This appears to be due to anonymous abject used in your code. Kindly ensure to improve upon this aspect.

**Error whiles adding object:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Rachel & Ouail, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦   🏃‍♂️ 🏃‍♀️  

## Highlights 👍  
- Correctly refactored objects, arrays, and or properties into a class!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

**You can also consider:**
- Kindly include details of the other co-contributor(s) in your ReadMe.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js [hunk 2]
```diff
@@ -1,52 +1,67 @@
...
-    <h3>${author}</h3>
-    <hr>`;
-  const deleteBtn = document.createElement('button');
-  deleteBtn.textContent = 'Remove';
-  li.insertBefore(deleteBtn, li.lastElementChild);
-  booksList.appendChild(li);
-  deleteBtn.addEventListener('click', () => {
-    books = books.filter((books) => {
-      if (books.title !== title && books.author !== author) {
+  addBook() {
```
Comment (Nemwel-Boniface) [inline script.js:12]: - [ ] [OPTIONAL] Even though this was not explicitly mentioned to you by the previous reviewer @rloterh , I strongly suggest you make this change as it was part of the project requirements. Kindly implement the local storage functionality when a user adds a book it persists in the browser's local storage. Likewise, when a user deletes a book it is also updated in the local storage.

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @Rachelwebdev        :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 2 above)
Comment (Rachelwebdev) [reply script.js:12]: Okay, I will implement this change in my code before merging. Thanks for the review and approval

File: script.js (hunk 1 above)
Comment (Rachelwebdev) [reply script.js]: Duly noted and changes would be implemented. Thanks for the review

Comment (Rachelwebdev) [reply script.js:16]: Thanks for the review.

**Pull request merged**